### ISSUE TITLE
Nexus needs to manage OPTE V2P mappings

### ISSUE THREAD

**Issue by jmpesp** [oxidecomputer/omicron#2290]
In OPTE, a `Virt2Phys` mapping describes the next hop of an encapsulated packet if that inner packet's destination remains in the VPC (otherwise, the destination is boundary services). OPTE currently creates a single `Virt2Phys` mapping when creating an xde device:

https://github.com/oxidecomputer/opte/blob/415bf265fad9af85e2fb5fcb2644c7be8f48b3ca/xde/src/xde.rs#L517-L530

The control plane currently does not set any other `Virt2Phys` mappings. As a result, if VMs in the same VPC are scheduled onto different physical host sleds they cannot communicate with each other - OPTE drops packets when there isn't a mapping for them:

https://github.com/oxidecomputer/opte/blob/415bf265fad9af85e2fb5fcb2644c7be8f48b3ca/oxide-vpc/src/engine/overlay.rs#L250-L304

A `Virt2Phys` mapping requires the following information:

- a VM's NIC's private IP address
- a VM's NIC's mac
- the underlay ipv6 address of the sled hosting the VM
- the VM's VPC VNI

Let's say there are two sleds, both with one VM each, where each VM has one NIC:

```
sled A: fd00:1122:3344:100::1
  VM A:
    NIC A: 172.30.0.5 / A8:40:25:F2:D4:4C

sled B: fd00:1122:3344:200::1
  VM B:
    NIC B: 172.30.0.6 / A8:40:25:FD:6E:AE
```

For these VMs to be able to communicate, two mappings need to be made:

1. on sled B, a mapping to NIC A: `pfexec /opt/oxide/opte/bin/opteadm set-v2p 172.30.0.5 A8:40:25:F2:D4:4C fd00:1122:3344:100::1 <VPC VNI>`

2. on sled A, a mapping to NIC B: `pfexec /opt/oxide/opte/bin/opteadm set-v2p 172.30.0.6 A8:40:25:FD:6E:AE fd00:1122:3344:200::1 <VPC VNI>`

Once this is done, the VMs can communicate.

It should be Nexus' responsibility to create these `Virt2Phys` mappings:

- when an instance is created, OPTE will create a mapping when each xde device is created, but Nexus will be responsible for all other mapping creation
- these mappings need to be updated when an instance is migrated
- these mappings need to be deleted when an instance is destroyed (this requires https://github.com/oxidecomputer/opte/issues/332)
- any creation or deletion of NICs will also necessitate updating these mappings

There's an unfortunate but necessary combinatorial explosion here. Ultimately, if a sled is hosting VMs in a VPC, a `Virt2Phys` mapping needs to exist for each destination NIC in that same VPC where the NIC is attached to a VM hosted on another sled. Something like:

```
for vpc in vpcs:
  for (sledA, sledB) in <iterator that yields all pairs of sleds currently hosting VMs in a VPC>(vpc):
    for vm hosted on sledB:
      for nic in vm.nics:
        # create a mapping to each nic for each vm hosted on sledB
        sledA.set_v2p(nic.private_ip, nic.mac, sledB.underlay_ipv6, vpc.vni)

    for vm hosted on sledA:
      for nic in vm.nics:
        # create a mapping to each nic for each vm hosted on sledA
        sledB.set_v2p(nic.private_ip, nic.mac, sledA.underlay_ipv6, vpc.vni)
```

**Comment by jmpesp** [comment]
I'm closing this - follow-up work is tracked in a separate issue.
